ambry-cloud/src/main/java/com/github/ambry/cloud/azure/AzureBlobDataAccessor.java

```diff
@@ -90,35 +92,34 @@ public AzureBlobDataAccessor(CloudConfig cloudConfig, AzureCloudConfig azureClou
 
   /**
    * Test constructor
-   * @param blobServiceClient the {@link BlobServiceClient} to use.
-   * @param blobBatchClient the {@link BlobBatchClient} to use.
+   * @param blobServiceAsyncClient the {@link BlobServiceAsyncClient} to use.
+   * @param blobBatchAsyncClient the {@link BlobBatchAsyncClient} to use.
    * @param clusterName the cluster name to use.
    * @param azureMetrics the {@link AzureMetrics} to use.
    * @param azureCloudConfig {@link AzureCloudConfig} object.
    */
-  AzureBlobDataAccessor(BlobServiceClient blobServiceClient, BlobBatchClient blobBatchClient, String clusterName,
-      AzureMetrics azureMetrics, AzureCloudConfig azureCloudConfig) {
+  AzureBlobDataAccessor(BlobServiceAsyncClient blobServiceAsyncClient, BlobBatchAsyncClient blobBatchAsyncClient,
```

Comment:
So we are replacing sync implementations with async implementations here. Would it be better to allow a config (maybe an existing one) to choose the implementation between sync or async?

Reply:
Yeah, since we would have moving the callers of StorageClient to be async as well, we are not having any cases where sync methods are being used. Hence removed them. In case there is a use case where sync functionality is needed, may be we can have callers of async client wait on the response?